virttest/openvswitch.py

```diff
@@ -469,30 +472,72 @@ def init_db(self):
         process.run('%s %s %s %s %s' %
                     (path.find_command("ovsdb-server"),
                      "--remote=punix:%s" % (self.db_socket),
-                     "--remote=db:Open_vSwitch,manager_options",
+                     "--remote=db:Open_vSwitch,Open_vSwitch,manager_options",
                      "--pidfile=%s" % (self.db_pidfile),
-                     "--detach"))
+                     "--detach --log-file %s") % (self.db_path))
         self.ovs_vsctl(["--no-wait", "init"])
 
     def start_ovs_vswitchd(self):
+        """
+        Start ovs vswitch daemon with only 1 socket
+        """
+        self.ovs_vsctl(["--no-wait", "set", "Open_vSwitch",
+                        ".", "other_config:dpdk-init=true"])
+        self.ovs_vsctl(["--no-wait", "set", "Open_vSwitch",
+                        ".", "other_config:dpdk-socket-mem='1024'"])
+        self.ovs_vsctl(["--no-wait", "set", "Open_vSwitch",
+                        ".", "other_config:dpdk-lcore-mask='0x1'"])
         process.run('%s %s %s %s' %
                     (path.find_command("ovs-vswitchd"),
                      "--detach",
                      "--pidfile=%s" % self.ovs_pidfile,
                      "unix:%s" % self.db_socket))
 
+    def create_bridge(self, br_name):
+        """
+        Create bridge
+
+        :param br_name: name of bridge
+        """
+        self.ovs_vsctl(["add-br", br_name, "-- set bridge", br_name,
+                        "datapath_type=netdev"])
+
+    def add_ports(self, br_name, port_names):
+        """
+        Add ports into bridge
+
+        :param br_name: name of bridge
+        :param port_names: port names split by space
+        """
+        for port_name in port_names.split():
+            self.ovs_vsctl(["add-port", br_name, port_name, "-- set Interface",
+                            port_name, "type=dpdkvhostuser"])
+            process.run('chown qemu:qemu %s/%s' % (self.pid_files_path, port_name))
```

Comment:
add shell=true for process.run()

Reply:
It can work well and no need to add it.

Reply:
it may cause some unexpected issue on some other linux distribution, suggest add shell=true

Reply:
updated